xla/stream_executor/rocm/BUILD

```diff
@@ -146,6 +146,7 @@ cc_library(
         "@com_google_absl//absl/functional:any_invocable",
         "@com_google_absl//absl/strings",
         "//xla/stream_executor",
+        "//xla/stream_executor:kernel",
```

Comment:
I think we can depend on `//xla/stream_executor:stream_executor_impl` instead. Otherwise, I think we should still keep private visibility for `//xla/stream_executor:kernel` and add an exception for `rocm_executor` target.

@ezhulenev Thoughts?

Reply:
We actually going in other direction, trying to make SE a "normal" library: (1) delete `_impl` (2) export targets for different functionalities, kernel, stream, device_memory, etc.